### PR TITLE
Adding springboot 2.3+ support

### PR DESCRIPTION
**Pull request by philippeagra**

### CODE REVIEW

**Comment by philippeagra** [comment]
fix https://github.com/ekino/ekino-tools-version/issues/72

File: app/services/SpringBootVersionService.scala [hunk 1]
```diff
@@ -18,17 +19,32 @@ import scala.xml.XML
...
+  private val groupName = """\s+group\("([^"]+)".*""".r
+  private val artefactName = """\s+"([^"]+)".*""".r
+  private val endBlock = """\t}""".r
+  private val innerEndBlock = """\t\t}""".r
+
+  private val baseSpringbootUrl = "https://raw.githubusercontent.com/spring-projects/spring-boot/v"
   private val pomPath = "/spring-boot-dependencies/pom.xml"
+  private val gradlePath = "/spring-boot-dependencies/build.gradle"
+  private val mavenVersions = List("2.0", "2.1", "2.2")
+  private val gradleVersions = List("2.3", "2.4", "2.5", "2.6", "2.7")
```
Comment (clemstoquart) [inline app/services/SpringBootVersionService.scala:33]: Not sure about the naming here. These look like spring boot versions.

File: app/services/SpringBootVersionService.scala [hunk 2]
```diff
@@ -18,17 +19,32 @@ import scala.xml.XML
   */
 object SpringBootVersionService {
 
-  private val propertyRegex = """\$\{(.*)\}""".r
-  private val baseSpringbootPomUrl = "https://raw.githubusercontent.com/spring-projects/spring-boot/v"
+  private val propertyRegex = """\$\{(.*)}""".r
+  private val libraryVersion = """\s+library\("[^"]+", "([^"]+)".*""".r
+  private val groupName = """\s+group\("([^"]+)".*""".r
+  private val artefactName = """\s+"([^"]+)".*""".r
```
Comment (celcius112) [inline app/services/SpringBootVersionService.scala:25]: small detail: `artifactName` (same typo appears a bit further down)

File: app/services/SpringBootVersionService.scala (hunk 1 above)
Comment (celcius112) [reply app/services/SpringBootVersionService.scala:33]: I think he wanted to mean that starting Spring Boot 2.3 the POM switch from Maven to Gradle. 

However do we really want to fix versions for this ? Do you want to add explicit support for each Spring boot versions ?

Comment (philippeagra) [reply app/services/SpringBootVersionService.scala]: @clemstoquart hello, yep the meaning was springboot version using maven or gradle. I renamed it springboot2VersionWithPom
@celcius112 springboot 2.3+ is the new default as the structure of build.gradle in springboot seems unchanged.

File: app/services/SpringBootVersionService.scala (hunk 2 above)
Comment (philippeagra) [reply app/services/SpringBootVersionService.scala]: All artefacts have been removed.

**Review by clemstoquart — APPROVED** [review]

**Pull request merged**